pyrex.py

```diff
@@ -59,6 +59,15 @@ def __init__(self, *args, **kwargs):
         self.optionxform = lambda option: option
 
 
+class ParsingError(Exception):
+    def __init__(self, message):
+        super().__init__(message)
+        self.message = message
```

Comment:
This is unnecessary. The message passed to `Exception` is the default when the exception is printed as a string:

```python
try:
    raise Exception("Hello")
except Exception as e:
    print("%s" % e)
```
``` 
Hello
```

Reply:
I'd probably just do:

```python
class ParsingError(Exception):
    pass
```

Reply:
Oh, bad luck on my part. I tested all of this using `KeyError` as my dummy exception type to carry along the string. I didn't like that it artificially enclosed the string inside single-quotes. Turns out that behavior is specific to `KeyError`. You're right that `str(Exception('Hello'))` does act the way I'd want.

Thanks. I'll update it.